helix-term/src/ui/editor.rs

```diff
@@ -175,7 +175,7 @@ impl EditorView {
 
                     // iterate over range char by char
                     for grapheme in RopeGraphemes::new(text) {
-                        if grapheme == "\n" {
+                        if grapheme == "\n" || grapheme == "\r\n" {
```

Comment:
Actually, this is a great place to use the `str_to_line_ending()` function I commented on above.  The drawing code should handle all line endings, including VT etc.  So here we could do something like:

```if str_to_line_ending(grapheme).is_some() {```

Reply:
Right, will do. Do all the line endings have the same meaning though? FF and VT for example seem to move to a new line, but also indent it. Wouldn't we have to match specific line endings then in places like this?

Reply:
That's a good point.  For now, I think we can get away with the simple solution I gave above just to make sure we're handling all the same line breaks as Ropey.  But we'll probably want to revisit it for special handling like that.

So for now, I'd say use `str_to_line_ending()` and add a TODO comment.

Reply:
Is rope_slice_to_line_ending needed? What if we just called str_to_line_ending(rope_slice.as_str().unwrap()) where rope_slice_to_line_ending is being used?

Reply:
That unwrap could panic if the line ending spans through two chunks. Though ropey ensures that doesn't happen I think.

Reply:
> 
> 
> That unwrap could panic if the line ending spans through two chunks. Though ropey ensures that doesn't happen I think.

I already deleted the comment lol. I still don't get why we are matching Crlf in both functions though.

Reply:
And in this case I think I want to use rope_slice_to_line_ending(&grapheme).is_some() right?

Reply:
> Do we need rope_slice_to_line_ending? Could we just use str_to_line_ending(rope_slice.as_str().unwrap()) instead where rope_slice_to_line_ending is being used?

That's a great point.  However, the thing I'd be a little concerned about is the potential to confuse people reading the code later: the invariant that Ropey never splits line break graphemes across chunks isn't obvious unless you already know about it.  Putting this in a separate function gives us the opportunity to document that we're taking advantage of that, so that it's obvious to everyone, and we hopefully avoid mistakes in the future.  For example, maybe someone looks at that code and thinks they can use the same pattern for other things.

Reply:
(But in general, I agree with you.  Getting rid of unneeded separate functions is generally a win IMO.)